### PR TITLE
Improve performance of `realm.create()` when running in node

### PR DESCRIPTION
**Pull request by tgoyne**
In total these changes cut the runtime of it by ~25% on the test case I've been measuring.

### CODE REVIEW

File: src/js_object_accessor.hpp [hunk 1]
```diff
@@ -68,10 +68,10 @@ class NativeAccessor {
 
     OptionalValue value_for_property(ValueType dict, std::string const& prop_name, size_t prop_index) {
         ObjectType object = Value::validated_to_object(m_ctx, dict);
-        if (!Object::has_property(m_ctx, object, prop_name)) {
+        ValueType value = Object::get_property(m_ctx, object, prop_name);
```
Comment (kneth) [inline src/js_object_accessor.hpp]: We could easily apply this pattern many other places 😄

File: src/node/node_string.hpp [hunk 2]
```diff
@@ -30,16 +30,35 @@ class String<node::Types> {
...
     operator v8::Local<v8::String>() const {
         return Nan::New(m_str).ToLocalChecked();
     }
 };
 
+inline String<node::Types>::String(const v8::Local<v8::String> &s) {
+    if (s.IsEmpty() || s->Length() == 0) {
+        return;
+    }
+    if (s->IsOneByte()) {
```
Comment (kneth) [inline src/node/node_string.hpp]: `ContainsOnlyOneByte` might be safer (but slower)

File: src/js_object_accessor.hpp (hunk 1 above)
Comment (kneth) [reply src/js_object_accessor.hpp]: But now we cannot know if we have the property or the value in `undefined`.

File: src/node/node_string.hpp (hunk 2 above)
Comment (tgoyne) [reply src/node/node_string.hpp]: No, it isn't safer. `IsOneByte()` checks the type of the string, while  `ContainsOnlyOneByte()` checks the type of the string, and then if it isn't one-byte then checks the contents of the string to see if it actually could be one-byte. As such, `IsOneByte()` has false negatives, not false positives.

Calling `Utf8Length()` after verifying it's the same thing as `Length()` is just a pointless pessimization.

File: src/js_object_accessor.hpp (hunk 1 above)
Comment (tgoyne) [reply src/js_object_accessor.hpp]: Drawing a distinction between no value and `undefined` is pretty strange for JS to begin with, and most of the API already didn't. I don't think this is a behavioral change to any existing working code; it's purely a widening in what sorts of objects we accept to create().

File: src/node/node_string.hpp (hunk 2 above)
Comment (tgoyne) [reply src/node/node_string.hpp]: Okay, I see why the test was failing without these changes: internal and external one-byte strings are actually different in v8, and the assumption that `Length() == Utf8Length()` only holds for external one-byte strings.

I think trying to do different things for one-byte and two-byte strings isn't worth it given this.

**Pull request merged**